packages/@react-aria/tooltip/src/useTooltipTrigger.ts

```diff
@@ -25,57 +26,73 @@ interface TooltipTriggerAria {
 
 export function useTooltipTrigger(props: TooltipTriggerAriaProps, state: TooltipTriggerState, ref: RefObject<HTMLElement>) : TooltipTriggerAria {
   let {
-    tooltipProps = {} as TooltipProps,
-    triggerProps = {} as TriggerProps,
     isDisabled
   } = props;
 
-  let tooltipId = useId(tooltipProps.id);
-  let triggerId = useId(triggerProps.id);
+  let tooltipId = useId();
+
+  let isHovered = useRef(false);
+  let isFocused = useRef(false);
+
+  let handleShow = () => {
+    if (isHovered.current || isFocused.current) {
+      state.open();
+    }
+  };
+
+  let handleHide = () => {
+    if (!isHovered.current && !isFocused.current) {
+      state.close();
+    }
+  };
 
   let onKeyDownTrigger = (e) => {
     if (ref && ref.current) {
       // dismiss tooltip on esc key press
       if (e.key === 'Escape') {
         e.preventDefault();
-        e.stopPropagation();
-        state.setOpen(false);
+        state.close();
       }
     }
   };
-
-  // abstract away knowledge of timing transitions from aria hook
-  let tooltipManager = state.tooltipManager;
-
-  let handleDelayedShow = () => {
-    if (isDisabled) {
-      return;
-    }
-    let triggerId = ref.current.id;
-    tooltipManager.showTooltipDelayed(state, triggerId);
+  let onHoverStart = () => {
+    isHovered.current = true;
+    handleShow();
   };
-
-  let handleDelayedHide = () => {
-    tooltipManager.hideTooltipDelayed(state);
+  let onHoverEnd = () => {
+    isHovered.current = false;
+    handleHide();
+  };
+  let onFocus = () => {
+    isFocused.current = true;
+    handleShow();
+  };
+  let onBlur = () => {
+    isFocused.current = false;
+    handleHide();
   };
 
   let {hoverProps} = useHover({
     isDisabled,
-    onHoverStart: handleDelayedShow,
-    onHoverEnd: handleDelayedHide
+    onHoverStart,
+    onHoverEnd
   });
 
+  let {focusableProps} = useFocusable({
+    isDisabled,
+    onFocus,
+    onBlur,
+    onKeyDown: onKeyDownTrigger
+  }, ref);
+
   return {
     triggerProps: {
-      id: triggerId,
       'aria-describedby': state.open ? tooltipId : undefined,
-      onKeyDown: chain(triggerProps.onKeyDown, onKeyDownTrigger),
-      ...hoverProps,
-      onFocus: handleDelayedShow,
-      onBlur: handleDelayedHide
+      ...mergeProps(focusableProps, hoverProps)
     },
     tooltipProps: {
       id: tooltipId
+      // should i keep the tooltip open if it gets hovered? (not the trigger, the tooltip itself)
```

Comment:
I'll see what I can figure out, though is v2 good enough reason to do that behavior in this case?

Reply:
good question, I don't know the history of why it was added to v2 cc @dannify @devongovett. Perhaps a Spectrum question?

Reply:
native tooltips don't seem to do this. I think it was true in v2 because we had one OverlayTrigger for everything (e.g. popovers). Tooltips are not interactive so should be ok to hide them when you move off of the trigger.